### PR TITLE
linkname: remove usage of net.lookupStaticHost

### PR DESCRIPTION
**Pull request by mmatczuk**
Relates to #895

### CODE REVIEW

**Review by alexh-sauce — APPROVED** [review]

File: http_proxy_test.go [hunk 1]
```diff
@@ -109,3 +109,37 @@ func TestNopDialer(t *testing.T) {
 		t.Fatalf("expected %v, got %v", nopDialerErr, err)
 	}
 }
+
+func TestIsLocalhost(t *testing.T) {
+	cfg := DefaultHTTPProxyConfig()
+	p, err := NewHTTPProxy(cfg, nil, nil, nil, stdlog.Default())
```
Comment (Choraden) [inline http_proxy_test.go:115]: May we have a test for a handler mode too (`newHTTPProxy`)?

Comment (Choraden) [reply http_proxy_test.go:115]: That suite should work without reading hosts file.

**Comment by Choraden** [comment]
Other than that - LGTM

File: http_proxy_test.go (hunk 1 above)
Comment (mmatczuk) [reply http_proxy_test.go:115]: We can do that in a separate issue. It uses convention in this file.

**Pull request merged**